reliability/single-fit/rum/src/test/kotlin/com/datadog/android/rum/integration/GlobalRumMonitorTest.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.integration
+
+import com.datadog.android.api.SdkCore
+import com.datadog.android.core.stub.StubSDKCore
+import com.datadog.android.rum.GlobalRumMonitor
+import com.datadog.android.rum.Rum
+import com.datadog.android.rum.RumConfiguration
+import com.datadog.android.rum.integration.tests.elmyr.RumIntegrationForgeConfigurator
+import com.datadog.android.rum.integration.tests.utils.MainLooperTestConfiguration
+import com.datadog.android.tests.assertj.StubEventsAssert.Companion.assertThat
+import com.datadog.tools.unit.annotations.TestConfigurationsProvider
+import com.datadog.tools.unit.extensions.TestConfigurationExtension
+import com.datadog.tools.unit.extensions.config.TestConfiguration
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mockito.mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(TestConfigurationExtension::class)
+)
+@ForgeConfiguration(RumIntegrationForgeConfigurator::class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class GlobalRumMonitorTest {
+
+    private lateinit var stubSdkCore: StubSDKCore
+
+    @StringForgery
+    private lateinit var fakeApplicationId: String
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        stubSdkCore = StubSDKCore(forge)
+        val fakeRumConfiguration = RumConfiguration.Builder(fakeApplicationId)
+            .trackNonFatalAnrs(false)
+            .build()
+        Rum.enable(fakeRumConfiguration, stubSdkCore)
+    }
+
+    @Test
+    fun `M return true W isRegistered()`() {
+        // Given
+
+        // When
+        val isRegistered = GlobalRumMonitor.isRegistered(stubSdkCore)
+
+        // Then
+        assertThat(isRegistered).isTrue()
+    }
+
+    @Test
+    fun `M return false W isRegistered() {sdkCore without RUM}`() {
+        // Given
+        val otherSdkCore: SdkCore = mock()
+
+        // When
+        val isRegistered = GlobalRumMonitor.isRegistered(otherSdkCore)
+
+        // Then
+        assertThat(isRegistered).isFalse()
+    }
+
+    companion object {
+        private val mainLooper = MainLooperTestConfiguration()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ Can we please add more information here, why do we need to prepare the MainLooper for this test ?

Reply:
This is used to ensure we can initialise the RUM feature in a JVM environement (where there's no real Main Looper)